model-optimizer/extensions/ops/one_hot.py

```diff
@@ -56,10 +58,15 @@ def infer(node: Node):
             assert node.has_valid('depth'), 'The node "{}" must have attribute "depth"'.format(node.name)
             depth = node.depth
         else:
-            assert_msg = "OneHot `{0}` ({1} input port value) should be scalar: node: `{2}`, {0} value: `{3}`"
+            assert_msg = "OneHot `{0}` (1 input port value) should be scalar: node: `{1}`, {0} value: `{2}`"
             depth = node.in_port(1).data.get_value()
-            assert depth is not None and depth.ndim == 0, assert_msg.format('depth', '1', node.name, depth)
-            depth = depth.item(0)
+            assert depth is not None, assert_msg.format('depth', node.name, depth)
+            if depth.ndim == 0:
```

Comment:
You can check that `depth.size == 1` instead of checks that you implemented below and always use `depth.item(0)`